jaggeryapps/dashboard/directories/js/directory.js

```diff
@@ -34,7 +34,31 @@ function addOrUpdateUserDirectory() {
         data: data,
     })
         .done(function (data) {
-            window.location.href = DIRECTORY_LIST_PATH;
+
+            var resp = $.parseJSON(data);
+            if (resp.success == true) {
+                window.location.href = DIRECTORY_LIST_PATH;
+            } else {
+
+                if (typeof resp.reLogin != 'undefined' && resp.reLogin == true) {
+                    window.top.location.href = window.location.protocol + '//' + serverUrl + '/dashboard/logout.jag';
```

Comment:
Can we define a config property for context (i.e dashboard) and not have hard coded dashboard preferences.